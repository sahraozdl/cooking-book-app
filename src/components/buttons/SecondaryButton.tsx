'use client';

import { Button } from '@headlessui/react';
import { type ButtonHTMLAttributes, type ReactNode } from 'react';
import clsx from 'clsx';

interface Props extends ButtonHTMLAttributes<HTMLButtonElement> {
  children: ReactNode;
}

export default function SecondaryButton({ children, className, ...props }: Props) {
  return (
    <Button
      {...props}
      className={clsx(
        'px-4 py-2 rounded-full border border-gray-300 text-gray-700 hover:bg-orange-300 transition bg-gray-200 cursor-pointer focus:outline-orange-500',
        className
      )}
    >
      {children}
    </Button>
  );
}
