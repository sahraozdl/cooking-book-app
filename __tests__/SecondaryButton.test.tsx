import { render, screen, fireEvent } from '@testing-library/react';
import SecondaryButton from '@/components/buttons/SecondaryButton';

describe('SecondaryButton', () => {
  it('renders children correctly', () => {
    render(<SecondaryButton>Cancel</SecondaryButton>);
    expect(screen.getByText('Cancel')).toBeInTheDocument();
  });

  it('handles click events', () => {
    const handleClick = jest.fn();
    render(<SecondaryButton onClick={handleClick}>Cancel</SecondaryButton>);
    fireEvent.click(screen.getByText('Cancel'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<SecondaryButton>Snapshot</SecondaryButton>);
    expect(asFragment()).toMatchSnapshot();
  });
});
