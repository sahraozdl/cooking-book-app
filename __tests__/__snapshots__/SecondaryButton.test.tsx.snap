// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`SecondaryButton matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="px-4 py-2 rounded-full border border-gray-300 text-gray-700 hover:bg-orange-300 transition bg-gray-200 cursor-pointer focus:outline-orange-500"
    data-headlessui-state=""
    type="button"
  >
    Snapshot
  </button>
</DocumentFragment>
`;
